### ISSUE TITLE
Bug: Sanitize gets called on every value

### ISSUE THREAD

**Issue by joshcummingsdesign** [stellarwp/validation#6]
In [Validator.php, line 59](https://github.com/stellarwp/validation/blob/develop/src/Validator.php#L59), the rule get spread into a closure. That means on each iteration, rules are appended to the array, a new array is not created.

```php
foreach ($ruleSets as $key => $rule) {
  ...
  $validatedRules[$key] = Config::getServiceContainer()->get(ValidationRuleSet::class)->rules(...$rule);
  ...
}
```

The closure is in [ValidationRuleSet.php, line 45](https://github.com/stellarwp/validation/blob/develop/src/ValidationRuleSet.php#L45).

```php
public function rules(...$rules): self
    {
        foreach ($rules as $rule) {
            if ($rule instanceof Closure) {
                $this->validateClosureRule($rule);
                $this->rules[] = $rule;
                ...
```

So, for example:

```php
$validator = new Validator([
    'licenses' => ['required'],
    'url' => ['required', 'striphtml']
], $values);
```

Adds the StripHTML rule to both `licenses` and `url`.

<img width="541" alt="Screen Shot 2023-02-03 at 5 57 57 PM" src="https://user-images.githubusercontent.com/11913201/216741272-2709196a-1ecb-43d6-af80-3a26e48398a0.png">

**Comment by JasonTheAdams** [comment]
@joshcummingsdesign I've tried to reproduce this via the `ValidatorTest::testRulesWithSanitizationAreApplied` test, and so far I've not been able to reproduce this. Regardless of the order I define the rules in, only the appropriate rules are applied to each value, and I did step through execution to verify that the `ValidationRuleSet` only contains the expected rules as well.

If you're able to reproduce this via a failing test that would be ideal and I'd be happy to quickly fix it.
